sdk/core/azure-core/src/main/java/com/azure/core/http/HttpHeaders.java

```diff
@@ -66,6 +66,34 @@ public int getSize() {
         return headers.size();
     }
 
+    /**
+     * Adds a {@link HttpHeader header} with the given name and value if a header with that name doesn't already exist,
+     * otherwise adds the {@code value} to the existing header.
+     *
+     * @param name The name of the header.
+     * @param value The value of the header.
+     * @return The updated HttpHeaders object.
+     */
+    public HttpHeaders add(String name, String value) {
```

Comment:
If this acts like an upsert, I think it would be clearer to name it `put()` or `set()`.

Reply:
`set` and `put` are already APIs on the class and act as `set` operations which is overriding the previously set value(s).